include/odp/api/spec/init.h

```diff
@@ -332,6 +332,20 @@ int odp_term_local(void);
  */
 int odp_term_global(odp_instance_t instance);
 
+/**
+ * Set thread specific log function
+ *
+ * By default, all ODP log writes use the global log function, which may be set
+ * as part of odp_init_t. Using this operation, an alternative ODP log function
+ * may be set for the calling thread. When set, ODP uses the thread specific log
+ * function for all log writes originating from ODP API calls made by the
+ * calling thread. Setting the log function to NULL causes the calling thread to
+ * use the global log function.
+ *
+ * @param func Log function
+ */
+void odp_log_thread_fn_set(odp_log_func_t func);
```

Comment:
Considering that this is API call, I think, it needs to be consistent with the API naming convention -- the function needs to show to which ODP module it belongs. Plus, to be consistent with the existing odp_log_func_t type, probably, the better function name would be -- odp_thread_log_func_set()? This is, of course, if we assume that it is a part of the thread module. If this is a part of the "initialization" module, where, it seems, more freedom exists with the naming (e.g. odp_init_..., odp_term_..., etc, then some rules for naming in that module needs to be specified in the API. This simplifies the work with the ODP documentation.

Reply:
It belongs to "ODP log" module, which is currently embedded into init.h. Log related definitions include: odp_log_level_t, odp_log_func_t and odp_override_log(). All these could be moved into a new odp_log.h file at some point. odp_override_log() could be renamed to odp_log_override(). It's likely not used much as it's just another way to set global log function (instead of setting it through init param). It could be also deprecated.